python/Ganga/Core/InternalServices/ShutdownManager.py

```diff
@@ -114,6 +114,10 @@ def add_priority(x):
 
     logger.info("Stopping Job processing before shutting down Repositories")
 
+    from Ganga.Core.GangaThread.WorkerThreads import _global_queues
+    if _global_queues:
+        _global_queues.lock()
```

Comment:
A comment explaining the reason for this would be very useful. Otherwise in the future it will be found and my code smell detector will trigger at the unbalanced `lock`/`unlock`.